core/services/pipeline/runner.go

```diff
@@ -180,7 +204,7 @@ func (r *runner) processRun() (anyRemaining bool, err error) {
 	ctx, cancel := utils.CombinedContext(r.chStop, r.config.JobPipelineMaxRunDuration())
 	defer cancel()
 
-	return r.orm.ProcessNextUnfinishedRun(ctx, r.ExecuteRun)
+	return r.orm.ProcessNextUnfinishedRun(ctx, r.executeRun)
```

Comment:
I think we need to actually not start all those goroutines reading from the pg notifs, otherwise they will also attempt to execute some run when a run is inserted (I think its impossible for them to actually execute it as the run will be complete always, but still we can skip the extra query)

Reply:
They are inserted in a finished_state - so they will never be picked up by the pipeline.

But they will generate a notification, we may need to modify that to only notify on insert of unfinished runs.

Reply:
yeah they won't be picked up, but we'd still hit the db looking for them

Reply:
The changed trigger fixes this